### PR TITLE
Fix SCORPIO CLASSIC build

### LINKED ISSUE #6066: Build error when using PIO_VERSION=1
Here is the output 
```
tail /compyfs/bish218/e3sm_scratch/f09_f09.I1850ELM/bld/spio.bldlog.231114-133119
gmake[2]: *** [pio/alloc_mod.F90] Error 127
/bin/sh: /qfs/people/bish218/projects/e3sm/e3sm-after-cime-submodule/externals/scorpio_classic/src/genf90/genf90.pl: No such file or directory
gmake[2]: *** Deleting file `pio/alloc_mod.F90'
gmake[2]: *** [pio/piodarray.F90] Error 127
gmake[2]: *** Deleting file `pio/piodarray.F90'
/bin/sh: /qfs/people/bish218/projects/e3sm/e3sm-after-cime-submodule/externals/scorpio_classic/src/genf90/genf90.pl: No such file or directory
gmake[2]: *** [pio/pio_msg_getput_callbacks.F90] Error 127
gmake[2]: *** Deleting file `pio/pio_msg_getput_callbacks.F90'
gmake[1]: *** [pio/CMakeFiles/pio.dir/all] Error 2
```

The build is looking for `genf90.pl` in the wrong place. 

Instead of 
```
/qfs/people/bish218/projects/e3sm/e3sm-after-cime-submodule/externals/scorpio_classic/src/genf90/genf90.pl
```
the file is at
```
/qfs/people/bish218/projects/e3sm/e3sm-after-cime-submodule/externals/scorpio_classic/pio/bin/genf90.pl
```

I created the case via:
```
cd cime/scripts
./create_newcase -case f09_f09.I1850ELM --compset I1850ELM --res f09_f09
cd f09_f09.I1850ELM
./xmlchange PIO_VERSION=1
./case.build
```

Here is the info about my
```
git show -n 1
commit f4fef0bf0c5778940c566d56e60d605b3a5ba0a9
Merge: c336779 8324c66
Author: Andrew M. Bradley <ambrad@cs.stanford.edu>
Date:   Tue Nov 14 11:24:11 2023 -0600

    Merge branch 'ambrad/homme/dp-pg' (PR #6057)

    Homme(xx)/pgN: Add pgN feature to doubly periodic planar mode.

    * Add pgN feature to planar doubly periodic mode.
    * Add some standalone-hommexx infrastructure to support using the same unit-test
      exe in multiple tests.
    * Add an init-planar capability to Hommexx unit tests.
    * Add a unit test: gllfvremap_planar_ut, planar version of old gllfvremap_ut
      unit test.
    * Add a BFB test for F90/C++ dycore consistency in planar mode with pg2 physics
      grid.

    e3sm_developer passes on Chrysalis. Additional tests for GPU/EAMxx pass on
    various machines.

    [BFB] except for new tests in homme_integration```
```

### PR DESCRIPTION
**Pull request by jgfouca**
Fixing SCORPIO CLASSIC (PIO_VERSION=1) builds

* Fixes location of genf90 for SCORPIO CLASSIC
* Also avoids ADIOS I/O type when using SCORPIO CLASSIC

Fixes #6066 

[BFB]

### CODE REVIEW

File: share/util/shr_pio_mod.F90 [hunk 1]
```diff
@@ -669,8 +669,10 @@ subroutine shr_pio_getiotypefromname(typename, iotype, defaulttype)
        iotype = pio_iotype_netcdf4p
     else if ( typename .eq. 'NETCDF4C') then
        iotype = pio_iotype_netcdf4c
+#ifdef PIO2
```
Comment (jgfouca) [inline share/util/shr_pio_mod.F90:672]: @jayeshkrishna , this particular code may have been broken for a while, but I think this easy fix should be what we want.

Comment (jayeshkrishna) [reply share/util/shr_pio_mod.F90:672]: Yes, I agree

**Comment by jayeshkrishna** [comment]
@jgfouca I just updated the PR desc and title. Please feel free to update it as needed

**Comment by jgfouca** [comment]
OK, there's one file in MCT that does not work when `-DPIO2` is provided:
`driver-mct/main/seq_io_mod.F90`

There's a couple places where `#ifdef PIO2` contains code that does not build:

```
#ifdef PIO2
    if(present(set_fill)) then
       if(set_fill) lset_fill = PIO_FILL
    endif
#endif
```

and

```
#ifdef PIO2
             rcode = pio_set_fill(cpl_io_file(lfile_ind), lset_fill, old_set_fill)
#endif
```

@jayeshkrishna  recommended a workaround that I just pushed.

**Comment by rljacob** [comment]
@dqwu add a test to e3sm_integration that uses scorpio classic.

**Comment by jgfouca** [comment]
@dqwu , yes, unused options are not an error.

**Comment by jgfouca** [comment]
@jayeshkrishna , do you approve now?

**Comment by dqwu** [comment]
@jgfouca 
When I tested this PR with ne4 F case on ANL GCE workstations there is a build error in the MPAS code on ADIOS type.
```
bld/cmake-bld/framework/mpas_io.f90:270:41:

  270 |             pio_iotype = PIO_iotype_adios
      |                                         1
Error: Symbol 'pio_iotype_adios' at (1) has no IMPLICIT type; did you mean 'pio_iotype'?
```
I think mpas_io.f90 needs a similar fix applied to share/util/shr_pio_mod.F90.

**Comment by jgfouca** [comment]
@dqwu , looking at the code:

```
#ifdef HAVE_ADIOS
            pio_iotype = PIO_iotype_adios
            pio_mode = 0
#else
```

I was wondering why HAVE_ADIOS would be on if we are using pio1. Then I looked at the MPAS CMake:

```
list(APPEND CPPDEFS "-DMPAS_NO_LOG_REDIRECT" "-DUSE_PIO2" "-DHAVE_ADIOS" "-DMPAS_NO_ESMF_INIT" "-DMPAS_ESM_SHR_CONST" "-DMPAS_PERF_MOD_TIMERS" "${MODEL_FORMULATION}" "${FILE_OFFSET}" "${ZOLTAN_DEFINE}" "-D_MPI" "-DMPAS_NAMELIST_SUFFIX=${NAMELIST_SUFFIX}" "-DMPAS_EXE_NAME=${EXE_NAME}")
```

So it looks like MPAS just assumes PIO2 + ADIOS, which seems wrong. I can fix.

**Comment by jgfouca** [comment]
MPAS PIO stuff has been fixed. I added @jonbob as a reviewer

**Review by dqwu — APPROVED** [review]

**Comment by jgfouca** [comment]
Merged to next. I will merge this to master tomorrow if the dashboard looks OK and then start integrating https://github.com/E3SM-Project/E3SM/pull/6046

**Comment by dqwu** [comment]
> Merged to next. I will merge this to master tomorrow if the dashboard looks OK and then start integrating #6046

Thanks, I will let you know when #6046 is ready to be re-merged to next.

**Pull request merged**